### PR TITLE
qemu: revert a pair of upstream commits on i686

### PR DESCRIPTION
**Pull request by vcunat**
This fixes nixosTests.* on i686.  I don't understand this, just tried: https://github.com/NixOS/nixpkgs/pull/206163#issuecomment-1356355119

###### Things done

- Built on platform(s)
  - [x] i686-linux
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Testing...

We should probably report this upstream. I suggested this approach simply because I don't think qemu gives a huge amount of user control over MSRs.

**Comment by risicle** [comment]
I'm not able to reproduce this error, but am not sure what target to build to do so. Have tried building `pkgsi686Linux.tests.testers.nixosTest-example` (without this fix) which works for me, but I guess this is an i686 _guest_ on my x86_64 _host_?

**Comment by RaitoBezarius** [comment]
Related: https://lists.nongnu.org/archive/html/qemu-discuss/2022-10/msg00030.html

**Comment by vcunat** [comment]
Random example fixed for me: `nix-build -QA pkgsi686Linux.nixosTests.locate` (on x86_64-linux NixOS)

**Comment by RaitoBezarius** [comment]
Potential upstream fix: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=1739c7017fb1d759965dcbab925ff5980a5318cb

**Comment by risicle** [comment]
Hmm can't reproduce on a host CPU Xeon E-2356G running nixos 22.05 x86_64, so I don't think I can help much here.

File: pkgs/applications/virtualization/qemu/default.nix [hunk 1]
```diff
@@ -112,6 +112,18 @@ stdenv.mkDerivation rec {
       revert = true;
     })
   ]
+  ++ lib.optionals stdenv.isi686 [
+    (fetchpatch {
+      url = "https://gitlab.com/qemu-project/qemu/-/commit/37656470f67398dd10101d7d940d660053f60ff5.diff";
```
Comment (risicle) [inline pkgs/applications/virtualization/qemu/default.nix:117]:
```suggestion
      name = "revert-msr-core-thread-count-support.part-1.patch";
      url = "https://gitlab.com/qemu-project/qemu/-/commit/37656470f67398dd10101d7d940d660053f60ff5.diff";
```

File: pkgs/applications/virtualization/qemu/default.nix [hunk 2]
```diff
@@ -112,6 +112,18 @@ stdenv.mkDerivation rec {
...
+      sha256 = "sha256-JNldXdtKYbyH+ctT7EWru7wwQV2FjqfBbt5DTEUsmkY=";
+      revert = true;
+    })
+    (fetchpatch {
+      url = "https://gitlab.com/qemu-project/qemu/-/commit/62a44fddb24fec35a6baf7e2c52b0e935a5bfa90.diff";
```
Comment (risicle) [inline pkgs/applications/virtualization/qemu/default.nix:122]:
```suggestion
      name = "revert-msr-core-thread-count-support.part-2.patch";
      url = "https://gitlab.com/qemu-project/qemu/-/commit/62a44fddb24fec35a6baf7e2c52b0e935a5bfa90.diff";
```

**Comment by vcunat** [comment]
Recent-ish kernel on host solves this, so we should prefer that way.

**Pull request closed without merging**